odoo/addons/base/tests/test_res_partner.py

```diff
@@ -157,3 +157,17 @@ def test_partner_is_public(self):
         self.assertFalse(self.env.ref('base.public_user').active)
         self.assertFalse(self.env.ref('base.public_partner').active)
         self.assertTrue(self.env.ref('base.public_partner').is_public)
+
+    def test_user_id_with_parent(self):
+        Partner = self.env['res.partner']
+        company_1 = self.env['res.company'].create({'name': 'company_1'})
+        test_user = self.env['res.users'].create({
+            'name': 'This user',
+            'login': 'thisu',
+            'email': 'this.user@example.com',
+            'company_id': company_1.id,
+            'company_ids': [company_1.id]
```

Comment:
for better git history
```suggestion
            'company_ids': [company_1.id],
```